File: java/src/org/nachodb/ReplicationMasterStorage.java
package org.nachodb;

/**
 * Storage performing replication of changed pages to specified slave nodes.
 */
public interface ReplicationMasterStorage extends Storage { 
    /**
     * Get number of currently available slave nodes
     * @return number of online replication slaves
     */
    public int getNumberOfAvailableHosts();
}   
    

File: csharp/src/ReplicationMasterStorage.cs
namespace Volante
{
    /// <summary>
    /// Storage performing replication of changed pages to specified slave nodes.
    /// </summary>
    public interface ReplicationMasterStorage : Storage 
    { 
        /// <summary>
        /// Get number of currently available slave nodes
        /// </summary>
        /// <returns>number of online replication slaves</returns>        
        int GetNumberOfAvailableHosts();
    }  
}
    